SPECS/tdnf/fix-tests-for-azl.patch

```diff
@@ -0,0 +1,157 @@
+From 767337327c8987b776d5667ebb0751b10f9acc3f Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Wed, 11 Sep 2024 16:24:05 -0700
+Subject: [PATCH] fix tests for azl
+
+---
+ pytests/repo/setup-repo.sh      | 16 ++++++++--------
+ pytests/tests/test_cache.py     |  3 ++-
+ pytests/tests/test_priority.py  |  2 +-
+ pytests/tests/test_signature.py |  7 ++++++-
+ pytests/tests/test_srpms.py     |  2 +-
+ 5 files changed, 18 insertions(+), 12 deletions(-)
+
+diff --git a/pytests/repo/setup-repo.sh b/pytests/repo/setup-repo.sh
+index a19bd20..8b54881 100755
+--- a/pytests/repo/setup-repo.sh
++++ b/pytests/repo/setup-repo.sh
+@@ -65,8 +65,8 @@ mkdir -p -m 755 ${BUILD_PATH}/BUILD \
+ cat << EOF > ${TEST_REPO_DIR}/gpgkeydata
+      %echo Generating a key for repogpgcheck signatures
+      %no-protection
+-     Key-Type: default
+-     Subkey-Type: default
++     Key-Type: RSA
++     Subkey-Type: RSA
+      Name-Real: tdnf test
+      Name-Comment: tdnf test key
+      Name-Email: tdnftest@tdnf.test
+@@ -107,9 +107,9 @@ cp -r ${BUILD_PATH}/RPMS ${PUBLISH_SHA512_PATH}
+ mkdir -p ${PUBLISH_PATH}/keys
+ gpg --armor --export tdnftest@tdnf.test > ${PUBLISH_PATH}/keys/pubkey.asc
+ 
+-createrepo ${PUBLISH_PATH}
+-createrepo ${PUBLISH_SRC_PATH}
+-createrepo -s sha512 ${PUBLISH_SHA512_PATH}
++createrepo --compatibility ${PUBLISH_PATH}
++createrepo --compatibility ${PUBLISH_SRC_PATH}
++createrepo --compatibility -s sha512 ${PUBLISH_SHA512_PATH}
+ 
+ modifyrepo ${REPO_SRC_DIR}/updateinfo-1.xml ${PUBLISH_PATH}/repodata
+ check_err "Failed to modify repo with updateinfo-1.xml."
+@@ -127,7 +127,7 @@ cat << EOF > ${TEST_REPO_DIR}/yum.repos.d/photon-test.repo
+ name=basic
+ baseurl=http://localhost:8080/photon-test
+ #metalink=http://localhost:8080/photon-test/metalink
+-gpgkey=file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY
++gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY
+ gpgcheck=0
+ enabled=1
+ EOF
+@@ -137,7 +137,7 @@ cat << EOF > ${TEST_REPO_DIR}/yum.repos.d/photon-test-sha512.repo
+ name=basic
+ baseurl=http://localhost:8080/photon-test-sha512
+ #metalink=http://localhost:8080/photon-test-sha512/metalink
+-gpgkey=file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY
++gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY
+ gpgcheck=0
+ enabled=0
+ EOF
+@@ -146,7 +146,7 @@ cat << EOF > ${TEST_REPO_DIR}/yum.repos.d/photon-test-src.repo
+ [photon-test-src]
+ name=basic
+ baseurl=http://localhost:8080/photon-test-src
+-gpgkey=file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY
++gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY
```

Comment:
For the key paths, it might be easier to instead link/copy them in the `%check` section to match the expectations from the tests. Specs tend to be more friendly to view if we ever change the key name/path and my hope is it'll be harder to miss this workaround at that time.

Reply:
Yup, or just add a symbolic link before we run the tests.

Reply:
Symbolic link added in during check section.